avalon/tools/delegates.py

```diff
@@ -0,0 +1,71 @@
+from ..vendor.Qt import QtWidgets, QtCore
+from .. import io
+
+from .models import TreeModel
+
+
+class VersionDelegate(QtWidgets.QStyledItemDelegate):
+    """A delegate that display version integer formatted as version string."""
+
+    version_changed = QtCore.Signal()
+    first_run = False
+    lock = False
+
+    def _format_version(self, value):
+        """Formats integer to displayable version name"""
+        return "v{0:03d}".format(value)
+
+    def displayText(self, value, locale):
+        assert isinstance(value, int), "Version is not `int`"
+        return self._format_version(value)
+
+    def createEditor(self, parent, option, index):
+        item = index.data(TreeModel.ItemRole)
+        if item.get("isGroup"):
+            return
+
+        editor = QtWidgets.QComboBox(parent)
+
+        def commit_data():
+            if not self.first_run:
+                self.commitData.emit(editor)  # Update model data
+                self.version_changed.emit()   # Display model data
+        editor.currentIndexChanged.connect(commit_data)
+
+        self.first_run = True
+        self.lock = False
+
+        return editor
+
+    def setEditorData(self, editor, index):
+        if self.lock:
+            # Only set editor data once per delegation
+            return
+
+        editor.clear()
+
+        # Current value of the index
+        value = index.data(QtCore.Qt.DisplayRole)
+        assert isinstance(value, int), "Version is not `int`"
+
+        # Add all available versions to the editor
+        item = index.data(TreeModel.ItemRole)
+        parent_id = item['version_document']['parent']
+        versions = io.find({"type": "version", "parent": parent_id},
+                           sort=[("name", 1)])
```

Comment:
Maybe we could also clean up all the single quote `'` into double quote `"` in this PR ?
Avalon uses double quote, right ?

Reply:
Fine with me. Will take a peek at doing that tomorrow. :)

Reply:
Should be resolved with [this commit](https://github.com/getavalon/core/pull/440/commits/4c30973ee482fd749ebf041aa9223a32ef9d5905). Note however that I only went through everything in `avalon.tools` to keep this PR contained to the tools. :)

Reply:
> Maybe we could also clean up all the single quote ' into double quote " in this PR ?
Avalon uses double quote, right ?

Yes, thank you David for being vigilant. This is one of those "pebbles in a shoe" that's innocent on its own but distracts the reader (at least me) from the meaning of code.